src/PubSubClient.h

```diff
@@ -146,9 +146,9 @@ class PubSubClient : public Print {
     unsigned long lastInActivity;
     bool pingOutstanding;
     MQTT_CALLBACK_SIGNATURE;
-    uint32_t readPacket(uint8_t*);
+    size_t readPacket(uint8_t*);
     bool readByte(uint8_t* result);
-    bool readByte(uint8_t* result, uint16_t* index);
+    bool readByte(uint8_t* result, size_t* index);
```

Comment:
This does change the interface, however I get it is needed to support messages > 64k.